src/Navigation/Xamarin.Forms/Navigation.Xamarin.Forms/NavigationServiceExtensions.cs

```diff
@@ -214,6 +150,24 @@ public static class NavigationServiceExtensions
             return registrationInfo;
         }
 
+
+        /// <summary>
+        /// Adds the given condition to the registration.
+        /// Every time a associated ViewModel is requested, the condition is evaluated before showing it.
+        /// </summary>
+        /// <param name="registration"></param>
+        /// <param name="condition"><see cref="Func{TResult}" />Condition to evaluate before showing</param>
+        /// <returns>Registration info</returns>
+        public static NavigationRegistration WithCondition(
+            this NavigationRegistration registration,
+            Func<bool> condition)
+        {
+            registration.Condition = condition;
```

Comment:
I ask again: What if the condition is null? Is it checked somewhere else? Or does it throw? And shouldn't we do a null check here? Because of the naming of the method and so on.

Reply:
If `null` is assigned to the property, `() => true` is used instead.
This is done in the `Condition` property so we do not need to check anywhere else.